mono/mini/mini-exceptions.c

```diff
@@ -1141,7 +1141,7 @@ ves_icall_get_trace (MonoException *exc, gint32 skip, MonoBoolean need_file_info
 		}
 
 		mono_debug_free_source_location (location);
-		mono_array_setref_internal (res, i, sf);
+		mono_array_setref_internal (res, i - skip, sf);
```

Comment:
Why did this work previously ?

Reply:
I guess it never worked if `skip` was > 0, e.g. this crashes on regular mono:
```csharp
using System;
using System.Diagnostics;
using System.Runtime.CompilerServices;

namespace HelloWorld
{
    class Program
    {
        private static Exception InvokeException()
        {
            try
            {
                ThrowException();
                return null;
            }
            catch (Exception ex)
            {
                return ex;
            }
        }
        
        [MethodImpl(MethodImplOptions.NoInlining)]
        private static void ThrowException() => throw new Exception();
        
        static void Main(string[] args)
        {
            int skipFrames = 1;
            Exception ex = InvokeException();
            var exceptionStackTrace = new StackTrace(ex, skipFrames);
            Console.WriteLine(exceptionStackTrace);
        }
    }
}
```

Reply:
Presumably because in most cases it was called with zero. In non-netcore build all the default constructors pass `METHODS_TO_SKIP == 0`.